app/parse.py

```diff
@@ -1,15 +1,59 @@
-from dataclasses import dataclass
+import csv
+from dataclasses import dataclass, fields, astuple
+from typing import List
+
+import requests
+from bs4 import BeautifulSoup, Tag
+
+URL = "https://quotes.toscrape.com"
 
 
 @dataclass
 class Quote:
     text: str
     author: str
-    tags: list[str]
+    tags: List[str]
+
+
+PRODUCT_FIELDS = [field.name for field in fields(Quote)]
+
+
+def scrape_single_quote(quote: Tag) -> Quote:
+    tags = quote.select_one(".tags").text.split("\n")
+    clear_tags = [
+        tag.strip() for tag in tags
+        if tag.strip() and tag.strip() != "Tags:"
+    ]
+    return Quote(
+        text=quote.select_one(".text").text,
+        author=quote.select_one(".author").text,
+        tags=clear_tags,
+    )
+
+
+def scrape_all_quotes() -> list[Quote]:
```

Comment:
The type hint `list[Quote]` is not compatible with Python versions below 3.9. If you need to support older versions, consider using `List[Quote]` from the `typing` module instead.